### PR TITLE
chore: update multiple kubeconfig merge logic

### PR DESCRIPTION
**Pull request by vprashar2929**
This PR updates the merge logic for multiple kubeconfig's.

### CODE REVIEW

File: providers/kind/kind.sh [hunk 1]
```diff
@@ -90,7 +90,7 @@ _setup_kind() {
 	run kind create cluster --name="${KIND_CLUSTER_NAME}" -v6 --config="$KIND_CONFIG_YAML"
 
 	info "Generating kubeconfig"
-	kind get kubeconfig --name="${KIND_CLUSTER_NAME}" | sed 's/kind-kind/kind-kepler/' > "$KIND_KUBECONFIG"
```
Comment (SamYuan1990) [inline providers/kind/kind.sh:93]: are we save to remove sed here?

File: main.sh [hunk 2]
```diff
@@ -80,8 +81,6 @@ cluster_up() {
 cluster_down() {
 	"$CLUSTER_PROVIDER"_down
 	rm "$KUBECONFIG_ROOT_DIR/$KEPLER_KUBECONFIG"
-	export KUBECONFIG="$KUBECONFIG_ROOT_DIR"/config
```
Comment (SamYuan1990) [inline main.sh:83]: should we keep this ? if the cluster is down, we don't have reason to keep config file?

File: main.sh [hunk 3]
```diff
@@ -61,15 +61,16 @@ cluster_up() {
...
-	mv -f all-in-one-kubeconfig.yaml "$KUBECONFIG_ROOT_DIR"/config
-	export KUBECONFIG="$KUBECONFIG_ROOT_DIR"/config
+	mkdir -p "$(basename "$KUBECONFIG_ROOT_DIR")"
+	mv -f "$kubeconfig" "${KUBECONFIG_ROOT_DIR}/${KEPLER_KUBECONFIG}"
 
-	kubectl config --kubeconfig="$KUBECONFIG" set-cluster kind-kepler
+	kubeconfig="$KUBECONFIG_ROOT_DIR/config":$(find "$KUBECONFIG_ROOT_DIR" \
+		-type f -name "*config*" | tr '\n' ':')
+	kubeconfig=${kubeconfig%:}
+	KUBECONFIG=$kubeconfig kubectl config view --merge --flatten >all-in-one-kubeconfig.yaml
```
Comment (SamYuan1990) [inline main.sh:70]: if it's all in one and later, we shut down current cluster, will we remove the cluster access to tidy the all in one confg yaml file?

**Comment by SamYuan1990** [comment]
@vprashar2929 , great work! I will some questions here, if all those items are clarified I am good to merge it.

File: providers/kind/kind.sh (hunk 1 above)
Comment (vprashar2929) [reply providers/kind/kind.sh:93]: Yeah, it just replaces `kind-kind` with `kind-kepler`. I am not sure why we added this in the first place🤔

File: main.sh (hunk 2 above)
Comment (vprashar2929) [reply main.sh:83]: Missed it😅. Yes we can remove it

File: main.sh (hunk 3 above)
Comment (vprashar2929) [reply main.sh:70]: Yes, when we bring the current cluster down it cleans up the merged config.
When cluster is up:
```
-rw-r--r--@ 1 vibhuprashar  staff  12242 Nov 22 16:56 kubeconfig
-rw-r--r--  1 vibhuprashar  staff   5583 Nov 23 22:04 config-kepler
drwxr-x---  4 vibhuprashar  staff    128 Nov 23 22:04 cache
-rw-r--r--  1 vibhuprashar  staff  17730 Nov 23 22:05 config
```
When cluster is down:
```
-rw-r--r--@ 1 vibhuprashar  staff  12242 Nov 22 16:56 kubeconfig
drwxr-x---  4 vibhuprashar  staff    128 Nov 23 22:04 cache
-rw-r--r--  1 vibhuprashar  staff  12219 Nov 23 22:06 config
```

**Review by SamYuan1990 — APPROVED** [review]
LGTM

**Pull request merged**